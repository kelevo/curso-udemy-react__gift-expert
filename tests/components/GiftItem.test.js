import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe( 'Pruebas en <GiftItem />', () => {

    const title = 'Patrick';
    const url   = 'https://kelevo.com/kelevo.png';

    test( 'Debe de hacer match con el snapshot', () => {

        const { container } = render( <GifItem title={ title } url={ url } /> );
        expect( container ).toMatchSnapshot();

    });

    test( 'Debe de mostrar la imagen con el URL y ALT indicado', () => {

        render( <GifItem title={ title } url={ url } /> );
        // screen.debug();
        // expect( screen.getByRole( 'img' ).src ).toBe( url );
        const { src, alt } = screen.getByRole( 'img' );
        expect( src ).toBe( url );
        expect( alt ).toBe( title );

    })

    // test( 'Debe de mostrar el titulo en la card', () => {

    //     render( <GifItem title={ title } url={ url } /> );
    //     expect( screen.getByText( title ) ).toBeTruthy();

    // })

});