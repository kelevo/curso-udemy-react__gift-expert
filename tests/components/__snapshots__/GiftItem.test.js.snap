// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="grid-gallery__item"
  >
    <img
      alt="Patrick"
      class="grid-gallery__image"
      src="https://kelevo.com/kelevo.png"
    />
  </div>
</div>
`;
